types/musickit-js/MusicKit.API.d.ts

```diff
@@ -1,20 +1,1531 @@
 declare namespace MusicKit {
     /**
-     * @todo add types for Apple Music API Objects
+     * A resource—such as an album, song, or playlist.
+     * https://developer.apple.com/documentation/applemusicapi/resource
      */
     interface Resource {
-        attributes: {
+        id: string;
+        // (Required) Persistent identifier of the resource.
+        type: string;
+        // (Required) The type of resource.
+        href?: string;
+        // A URL subpath that fetches the resource as the primary object. This member is only present in responses.
+        attributes?: {
+            [key: string]: any;
+        };
```

Comment:
What do you think about using the `Record` type where you define maps?
It's be a bit more compact :)

```suggestion
        attributes?: Record<string, any>;
```